Packs/CheckPointHarmonyEndpoint/Integrations/CheckPointHarmonyEndpoint/CheckPointHarmonyEndpoint.py

```diff
@@ -0,0 +1,2797 @@
+        limit (int, optional): The maximum number of records to retrieve.
+
+    Raises:
+        ValueError: Appropriate error message.
+    """
+    if page_size and (page_size < MIN_PAGE_SIZE or page_size > MAX_PAGE_SIZE):
+        raise ValueError(
+            f"page_size argument must be greater than {MIN_PAGE_SIZE} and smaller than {MAX_PAGE_SIZE}."
+        )
+    if page and page < MIN_PAGE_NUM:
+        raise ValueError(f"page argument must be greater than {MIN_PAGE_NUM - 1}.")
+    if limit and limit <= MIN_LIMIT:
+        raise ValueError(f"limit argument must be greater than {MIN_LIMIT}.")
+
+
+def get_pagination_args(args: dict[str, Any]) -> tuple:
+    """Return the correct limit and offset for the API
+        based on the user arguments page, page_size and limit.
+
+    Args:
+        args (dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        Tuple: new_limit, offset, pagination_message.
+    """
+    page = arg_to_number(args.get("page"))
+    page_size = arg_to_number(args.get("page_size"))
+    limit = arg_to_number(args.get("limit"))
+
+    validate_pagination_arguments(page, page_size, limit)
+
+    new_page = 0
+    new_page_size = limit
+
+    if page and page_size:
+        new_page_size = page_size
+        new_page = page - 1
+
+    pagination_message = (
+        f"Showing page {new_page+1}.\nCurrent page size: {new_page_size}."
+    )
+
+    return new_page, new_page_size, pagination_message
+
+
+def validate_filter_arguments(column_name: str | None = None, filter_type: str = None):
+    """Validate filter arguments values are allowed.
+
+    Args:
+        column_name (str, optional): The column name to filter by. Defaults to None.
+        filter_type (str, optional): The filter operator. Defaults to None.
+
+    Raises:
+        ValueError: Raise error in case column_name or filter_type values are not allowed.
+    """
+    if column_name and column_name not in COLUMN_NAMES:
+        raise ValueError(
+            f"'column_name' must be one of the followings: {COLUMN_NAMES}."
+        )
+
+    if filter_type and filter_type not in FILTER_TYPES:
+        raise ValueError(
+            f"'filter_type' must be one of the followings: {FILTER_TYPES}."
+        )
+
+
+def extract_query_filter(args: dict[str, Any]) -> list[dict[str, Any]]:
+    """Extract query filters from the specified arguments.
+
+    Args:
+        args (dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        list[dict[str, Any]]: The updated query filter according to Harmony requirements.
+    """
+    query_filter = []
+
+    if filter_by_query := args.get("filter"):
+        queries = argToList(filter_by_query, "' , ")
+
+        for query in queries:
+            query_parts = query.split(" ")
+
+            if len(query_parts) != 3:
+                raise ValueError(
+                    "'filter_by_query' must be in the following format: 'column_name filter_type filter_value'."
```

Comment:
```suggestion
                    "'filter' must be in the following format: 'column_name filter_type filter_value'."
```
I think this is the argument name